include/manif/impl/sgal3/SGal3Tangent_base.h

```diff
@@ -0,0 +1,635 @@
+
+namespace manif {
+
+//
+// Tangent
+//
+
+/**
+ * @brief The base class of the SGal3 tangent.
+ */
+template <typename _Derived>
+struct SGal3TangentBase : TangentBase<_Derived> {
+private:
+
+  using Base = TangentBase<_Derived>;
+  using Type = SGal3TangentBase<_Derived>;
+
+public:
+
+  MANIF_TANGENT_TYPEDEF
+  MANIF_INHERIT_TANGENT_API
+  MANIF_INHERIT_TANGENT_OPERATOR
+
+  using LinBlock = typename DataType::template FixedSegmentReturnType<3>::Type;
+  using AngBlock = typename DataType::template FixedSegmentReturnType<3>::Type;
+  using ConstLinBlock = typename DataType::template ConstFixedSegmentReturnType<3>::Type;
+  using ConstAngBlock = typename DataType::template ConstFixedSegmentReturnType<3>::Type;
+
+  using Base::data;
+  using Base::coeffs;
+
+protected:
+
+  using Base::derived;
+
+  MANIF_DEFAULT_CONSTRUCTOR(SGal3TangentBase)
+
+public:
+
+  MANIF_TANGENT_ML_ASSIGN_OP(SGal3TangentBase)
+
+  // Tangent common API
+
+  /**
+   * @brief Hat operator of SGal3.
+   * @return An element of the Lie algebra se_2_3.
+   * @note See Eq. (169).
+   */
+  LieAlg hat() const;
+
+  /**
+   * @brief Get the SGal3 element.
+   * @param[out] -optional- J_m_t Jacobian of the SGal3 element wrt this.
+   * @return The SGal3 element.
+   * @note This is the exp() map with the argument in vector form.
+   */
+  LieGroup exp(OptJacobianRef J_m_t = {}) const;
+
+  /**
+   * @brief This function is deprecated.
+   * Please considere using
+   * @ref exp instead.
+   */
+  MANIF_DEPRECATED
+  LieGroup retract(OptJacobianRef J_m_t = {}) const;
+
+  /**
+   * @brief Get the right Jacobian of SGal3.
+   */
+  Jacobian rjac() const;
+
+  /**
+   * @brief Get the left Jacobian of SGal3.
+   */
+  Jacobian ljac() const;
+
+  /**
+   * @brief Get the small adjoint matrix ad() of SGal3
+   * that maps isomorphic tangent vectors of SGal3
+   * @return
+   */
+  Jacobian smallAdj() const;
+
+  // SGal3Tangent specific API
+
+  //! @brief Get the linear velocity part.
+  LinBlock lin();
+  const ConstLinBlock lin() const;
+
+  //! @brief Get the angular part.
+  AngBlock ang();
+  const ConstAngBlock ang() const;
+
+  //! @brief Get the linear acceleration part
+  LinBlock lin2();
+  const ConstLinBlock lin2() const;
+
+  Scalar t() const;
+
+public: /// @todo make protected
+
+  const Eigen::Map<const SO3Tangent<Scalar>> asSO3() const {
+    return Eigen::Map<const SO3Tangent<Scalar>>(coeffs().data()+6);
+  }
+
+  Eigen::Map<SO3Tangent<Scalar>> asSO3() {
+    return Eigen::Map<SO3Tangent<Scalar>>(coeffs().data()+6);
+  }
+
+  static void fillE(
+    Eigen::Ref<Eigen::Matrix<Scalar, 3, 3>> E,
+    const Eigen::Map<const SO3Tangent<Scalar>>& so3
+  );
+
+  // static void fillL(
+  //   Eigen::Ref<Eigen::Matrix<Scalar, 3, 3>> E,
+  //   const Eigen::Map<const SO3Tangent<Scalar>>& so3
+  // );
+
+  // static void fillS(
+    // Eigen::Ref<Eigen::Matrix<Scalar, 3, 3>> S,
+    // const Eigen::Map<const SO3Tangent<Scalar>>& so3,
+    // linearVel,
+    // t
+  // );
```

Comment:
remove these commented bits. Also, in case we want to keep them, `S` was renamed `N1` in the implementation of `ljac()`